src/services/redux/user/saga.js

```diff
@@ -0,0 +1,60 @@
+import {
+  call, takeLatest, put, fork,
+} from 'redux-saga/effects';
+import { workerSpecReceive } from 'services/redux/doctors/saga';
+import notify from 'services/notify';
+import { setToken } from '../../api/instance';
+import { USER_LOG_IN, USER_REGISTER } from './constants';
+import { getProfile, userLogin, userRegistration } from '../../api/auth/requests';
+import {
+  pendingUserAction, rejectedUserAction, userLoggedIn, userProfileUpdated, userRegistered,
+} from './actions';
+
+function* workerUserLogin({ payload: { values, navigate } }) {
+  const notification = notify.initial('Please wait...');
+  try {
+    yield notify.update(notification, 'pending', 'Processing...');
+    yield put(pendingUserAction());
+    const { data: { access_token } } = yield call(() => userLogin(values));
+    setToken(access_token);
+    const { data } = yield call(() => getProfile());
```

Comment:
не нужно дополнительных функций
```suggestion
    const { data } = yield call(getProfile);
```